app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -766,6 +791,64 @@ class SectionRenderingService(
       )
   }
 
+  private def findFileId(ei: ExtraInfo, formComponent: FormComponent): FileId =
+    ei.envelope
+      .find(formComponent.modelComponentId)
+      .map(_.fileId)
+      .getOrElse(ei.envelope.mapping.fileIdFor(formComponent.id))
+
+  private def isVisible(formComponent: FormComponent, ei: ExtraInfo): Boolean = formComponent.includeIf.fold(true) {
+    includeIf =>
+      ei.formModelOptics.formModelVisibilityOptics.evalIncludeIfExpr(includeIf, None)
+  }
+
+  private def htmlForUpscan(
+    renderUnit: RenderUnit,
```

Comment:
This should not use `renderUnit: RenderUnit`. It should use `formComponent: FormComponent` which we know is upscan "file"